### DISCUSSION TITLE
데이터 업데이트를 위한 방법 고민

### DISCUSSION THREAD

**Discussion by sphy1597** [wanted-7h/hotplace#12] [Q&A]
데이터를 주기적으로 업데이트 할 때 전체 데이터를 전부 비교하기에는 서버에 부담이 클것으로 예상되 방법을 고민중입니다!

1. 각 데이터 별로 해시함수를 적용해서 주기적으로 받아온 데이터와 해시값만 비교하여 업데이트 진행
2. 지역별로 or 업종 등의 카테고리로 나눠서 비교를 진행
3. 그 외 방식

**Comment by scarf005** [comment]
> 1. 각 데이터 별로 해시함수를 적용해서 주기적으로 받아온 데이터와 해시값만 비교하여 업데이트 진행

과제 요구사항에서는 `가게명` + `주소`를 Primary key로 해 DB에 저장하라고 되어 있으니, 해당 Primary Key를 기준으로 비교해보면 어떨까요?

**Comment by Yonge2** [comment]
공공데이터에서는 업종별로 제공해주는 것으로 알고 있는데,
동시에 모두 진행하는 것이 걱정 되신다면, 업종별로 시간대를 나누기는 어떠신가요?

예를 들어, 
1. 사용이 잘 안되는 시간대에 1시간 간격으로 순차 업뎃
2. 오늘은 중식, 내일은 한식 ... 

+ 만약에 받아온 데이터 중에, 최신 업데이트 날짜(데이터 생성 시간?)이 포함되어 있다면 앞 단계에서 먼저 그거 기준으로 업뎃 여부를 결정하는 것도 괜찮을 것 같아요!
